internal/controllers/phase_ensure_operator_group.go

```diff
@@ -59,7 +61,15 @@ func (r *AddonReconciler) reconcileOperatorGroup(
 		return fmt.Errorf("getting OperatorGroup: %w", err)
 	}
 
-	if !equality.Semantic.DeepEqual(currentOperatorGroup.Spec, operatorGroup.Spec) {
+	if preventResourceAdoption &&
+		len(currentOperatorGroup.OwnerReferences) == 0 ||
+		!hasEqualControllerReference(currentOperatorGroup, operatorGroup) {
+		return errNotOwnedByUs
+	}
+
+	if !equality.Semantic.DeepEqual(currentOperatorGroup.Spec, operatorGroup.Spec) ||
+		!equality.Semantic.DeepEqual(currentOperatorGroup.OwnerReferences,
+			operatorGroup.OwnerReferences) {
```

Comment:
Now that we can adopt `Namespace` objects, we should be able to adopt existing objects inside it not only by looking at the spec, but also the ownerReference to prevent any inconsistencies. For example, without this check, we could update the ownerRef on a namespace, but not on any other object unless there's a drift between the crrent and desired spec of the object itself.